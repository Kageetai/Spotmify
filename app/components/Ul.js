import styled from 'styled-components';

const Ul = styled.ul`
  ${props => props.clean ? `
    list-style: none;
    margin: 0;
    padding: 0;
  ` : null}
`;

export const Li = styled.li`
  ${props => props.clean ? `
    margin: 0;
    padding: 0;
  ` : null}
  ${props => props.inline ? `
    display: inline;
  ` : null}
`;

export default Ul;
