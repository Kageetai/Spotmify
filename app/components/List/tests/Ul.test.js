import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Ul from '../Ul';

describe('<Ul />', () => {
  it('should match the snapshot', () => {
    const wrapper = renderer.create(<Ul />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an <ul> tag', () => {
    const wrapper = mount(<Ul />);
    expect(wrapper.find('ul').type()).toEqual('ul');
  });

  it('should have a className attribute', () => {
    const wrapper = mount(<Ul />);
    expect(wrapper.find('ul').prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const wrapper = mount(<Ul id={id} />);
    expect(wrapper.find('ul').prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const wrapper = mount(<Ul attribute="test" />);
    expect(wrapper.find('ul').prop('attribute')).toBeUndefined();
  });
});
