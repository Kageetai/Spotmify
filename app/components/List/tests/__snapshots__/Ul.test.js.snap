// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ul /> should match the snapshot 1`] = `
.c0 {
  list-style: none;
  margin: 0;
  width: 100%;
  max-height: 30em;
  overflow-y: auto;
  padding: 0 1em;
}

<ul
  className="c0"
/>
`;
